tools/building.py

```diff
@@ -1358,9 +1358,19 @@ def strip(infile, outfile, debug=False, producers=False):
 def emit_debug_on_side(wasm_file, wasm_file_with_dwarf):
   # if the dwarf filename wasn't provided, use the default target + a suffix
   wasm_file_with_dwarf = shared.Settings.SEPARATE_DWARF
+  embedded_path = shared.Settings.SEPARATE_DWARF_URL
   if wasm_file_with_dwarf is True:
     wasm_file_with_dwarf = wasm_file + '.debug.wasm'
-  embedded_path = shared.Settings.SEPARATE_DWARF_URL or wasm_file_with_dwarf
+    # assume by default that the debug file is adjacent to the main one; to
+    # override that SEPARATE_DWARF_URL can be used
+    wasm_file_with_dwarf = os.path.basename(wasm_file_with_dwarf)
```

Comment:
Btw, in theory you don't need this separate code path - the one below with `relpath` should cover both cases (since basename is just a partial case of relpath). So the code could be simplified to something like:

```python
if wasm_file_with_dwarf is True:
  wasm_file_with_dwarf = wasm_file + '.debug.wasm'
if not embedded_path:
  embedded_path = os.path.relpath(wasm_file_with_dwarf, os.path.dirname(wasm_file))
```

Reply:
Good idea, done.